### PR TITLE
User can search by ingredient & view a list of recipes 

### PR DESCRIPTION
**Pull request by beccaburns**
[PT-Link](https://www.pivotaltracker.com/story/show/168978853)
[PT-Link](https://www.pivotaltracker.com/story/show/168979054)

## In this PR
- We created the feature and corresponding step definitions for this user story

## What we learned
- We generated a recipe controller with a haml file that presents the search bar where users can input ingredients and view recipes from the API
- We learned how to integrate the API through the module to the index to present the results needed
- Continued learning about how to write haml and corresponding styling
- Deeper understanding of paths and how data flows

LandingPage_NotSoTrivial

Screen Shot 2019-10-08 at 3 35 49 PM

Screen Shot 2019-10-08 at 3 36 02 PM

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class RecipesController < ApplicationController
+  def index
+    @recipes = FoodService.get_recipes(params[:search])
+  end
+
+  private
+
```
Comment (piavW) [inline app/controllers/recipes_controller.rb:7]: Please remove empty row

**Comment by faraznaeem** [comment]
MERGE conflicts!

**Comment by beccaburns** [comment]
> MERGE conflicts!

File: features/support/env.rb [hunk 2]
```diff
@@ -1,10 +1,26 @@
 require 'cucumber/rails'
+require 'webmock/cucumber'
 ActionController::Base.allow_rescue = false
+WebMock.allow_net_connect!
 
 begin
   DatabaseCleaner.strategy = :transaction
 rescue NameError
   raise "You need to add database_cleaner to your Gemfile (in the :test group) if you wish to use it."
 end
```
Comment (piavW) [inline features/support/env.rb:11]: Do you have database cleaner? No need to have the code-block if we don't. :)

**Review by piavW — APPROVED** [review]

**Review by AlexSaxena — APPROVED** [review]

**Review by yasmineezequiel — APPROVED** [review]

File: app/views/layouts/mailer.html.haml [hunk 3]
```diff
@@ -1,8 +1,6 @@
-!!!
 %html
   %head
-    %meta{content: "text/html; charset=utf-8", "http-equiv" => "Content-Type"}/
+    %meta{:content => "text/html; charset=utf-8", "http-equiv" => "Content-Type"}/
```
Comment (faraznaeem) [inline app/views/layouts/mailer.html.haml]: did you add back ha hash rocket?

File: features/user_can_sign_up_for_an_account.feature [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  Background: 
+    Given I visit the home page
+    And I click "Sign up here"
+
+  Scenario: Creating an account[Happy Path]
+    When I fill in "Email" with "berg@sten.com"
+    And I fill in "Password" with "password"
+    And I fill in "Password confirmation" with "password"
+    And I click "Sign up"
+    Then I should see "Logged in as berg@sten.com"
```
Comment (faraznaeem) [inline features/user_can_sign_up_for_an_account.feature]: This feature is out of scope

**Review by faraznaeem — CHANGES_REQUESTED** [review]

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**